src/dusk/serde_support.rs

```diff
@@ -0,0 +1,97 @@
+extern crate alloc;
+
+use alloc::string::String;
+
+use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{AffinePoint, ExtendedPoint};
+
+impl Serialize for AffinePoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s = hex::encode(self.to_bytes());
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for AffinePoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let s = String::deserialize(deserializer)?;
+        let decoded = hex::decode(&s).map_err(Error::custom)?;
+        let decoded_len = decoded.len();
+        let bytes: [u8; 32] = decoded
+            .try_into()
+            .map_err(|_| Error::invalid_length(decoded_len, &"32"))?;
+        AffinePoint::from_bytes(bytes)
+            .into_option()
+            .ok_or(Error::custom(
+                "Failed to deserialize AffinePoint: invalid AffinePoint",
+            ))
+    }
+}
+
+impl Serialize for ExtendedPoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        AffinePoint::from(self).serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ExtendedPoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        AffinePoint::deserialize(deserializer).map(Into::into)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use group::Group;
+    use rand_core::OsRng;
+
+    use crate::{AffinePoint, ExtendedPoint};
+
+    #[test]
+    fn serde_affine_point() {
+        let mut rng = OsRng;
+        let point = ExtendedPoint::random(&mut rng);
```

Comment:
It is better imo to use [SeedableRng](https://rust-random.github.io/rand/rand_core/trait.SeedableRng.html) here as well like in other tests you did.